docs/recipes/use-atom-effect.mdx

```diff
@@ -0,0 +1,66 @@
+---
+title: useAtomEffect
+nav: 8.03
+keywords: effect, atom effect, side effect, side-effect, sideeffect, hook, useAtomEffect
+---
+
+> `useAtomEffect` runs side effects in response to changes in atoms or props using [atomEffect](../extensions/effect.mdx).
+
+The effectFn reruns whenever the atoms it depends on change or the effectFn itself changes. Be sure to memoize the effectFn if it's a function defined in the component.
+
+```ts
+import { useMemo } from 'react'
+import { useAtomValue } from 'jotai/react'
+import { atomEffect } from 'jotai-effect'
+
+type CleanupFn = () => void
+
+type EffectFn = (
+  get: Getter & { peek: Getter },
+  set: Setter & { recurse: Setter },
+) => CleanupFn | void
+
+export function useAtomEffect(effectFn: EffectFn) {
+  useAtomValue(useMemo(() => atomEffect(effectFn), [effectFn]))
+}
+```
+
+### Example Usage
+
+```tsx
+import { atom, useAtom } from 'jotai'
+import { atomFamily } from 'jotai/utils'
+import { useAtomEffect } from './useAtomEffect'
+
+const channelSubscriptionAtomFamily = atomFamily<Channel>(
+  (channelId: string) => {
+    return atom(new Channel(channelId))
+  },
+)
+const messagesAtom = atom<Message[]>([])
+
+function Messages({ channelId }: { channelId: string }) {
+  const [messages] = useAtom(messagesAtom)
+  useAtomEffect(
+    useCallback(
```

Comment:
Maybe we should add a small note about useCallback/useMemo that they aren't for semantic guarantee by React. So, it **may** run the effect twice **in the future**. (But, if it really happens, we could just rewrite the recipe using useRef based memo/callback. For example: https://github.com/alexreardon/use-memo-one )

Reply:
Good point. I think the recipe should use the stable useCallback from use-memo-one and include this general note on useCallback.